src/App/contexts/taskContext.js

```diff
@@ -20,22 +22,60 @@ export const TaskProvider = (props) => {
     else setIsCompletedAll(true);
   };
 
+  // Filter data method
+  const getAll = () => {
+    // get all tasks
+    setFilteredData(taskData);
+    setActiveFilter('ALL');
+  };
+
+  const getActive = () => {
+    // get all active tasks
+    const activeTasks = taskData.filter((task) => task.isCompleted === false);
+    setFilteredData(activeTasks);
+    setActiveFilter('ACTIVE');
+  };
+
+  const getCompleted = () => {
+    // get all completed tasks
+    const completedTasks = taskData.filter((task) => task.isCompleted === true);
+    setFilteredData(completedTasks);
+    setActiveFilter('COMPLETED');
+  };
+
+  // Call API method
   async function fetchData() {
     try {
+      setIsLoading(true);
       const response = await axios.get(`${SERVER_URL}`);
-      setTaskData(
-        sortArray(response.data, {
-          by: 'createdAt',
-          order: 'desc',
-        }),
-      );
+      setIsLoading(false);
+      const sortedData = sortArray(response.data, {
+        by: 'createdAt',
+        order: 'desc',
+      });
+      setTaskData(sortedData);
+      switch (activeFilter) {
+        case 'ALL':
+          setFilteredData(sortedData);
+          break;
+        case 'ACTIVE':
+          setFilteredData(sortedData.filter((task) => task.isCompleted === false));
+          break;
+        case 'COMPLETED':
+          setFilteredData(sortedData.filter((task) => task.isCompleted === true));
+          break;
+        default:
+          setFilteredData(sortedData);
+          break;
+      }
     } catch (error) {
       console.log(error);
     }
   }
 
   async function postData(body) {
     try {
+      setIsLoading(true);
       await axios.post(`${SERVER_URL}`, body);
       fetchData();
     } catch (err) {
```

Comment:
What happens if there is an issue with the connection and throw err? Is the loading screen still loading forever?

Reply:
I'm sorry not covering this problem. I'll fix this right away

Reply:
I fixed this problem